### PR TITLE
Predict entitystorage

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

ooh yeah baby let's go let's predict this shit.
prediction coin stonks could be up even more but i don't want to touch the visualizer rn

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

no cl no fun

### CODE REVIEW

File: Content.Server/Storage/Components/EntityStorageComponent.cs [hunk 1]
```diff
@@ -1,97 +1,12 @@
...
+using Content.Shared.Storage.Components;
+using Robust.Shared.GameStates;
 
 namespace Content.Server.Storage.Components;
 
-[RegisterComponent]
-public sealed class EntityStorageComponent : Component, IGasMixtureHolder
+[RegisterComponent, NetworkedComponent, ComponentReference(typeof(SharedEntityStorageComponent))]
+public sealed class EntityStorageComponent : SharedEntityStorageComponent, IGasMixtureHolder
 {
```
Comment (metalgearsloth) [inline Content.Server/Storage/Components/EntityStorageComponent.cs:9]: You can put gasmixtures on shared I thought, also sidenote networkedcomponent is inheritable

File: Content.Shared/Lock/LockSystem.cs [hunk 2]
```diff
@@ -85,7 +85,7 @@ private void OnStorageOpenAttempt(EntityUid uid, LockComponent component, ref St
     {
         if (!component.Locked)
             return;
-        if (!args.Silent)
+        if (!args.Silent && _timing.IsFirstTimePredicted && _net.IsClient)
             _sharedPopupSystem.PopupEntity(Loc.GetString("entity-storage-component-locked-message"), uid);
```
Comment (metalgearsloth) [inline Content.Shared/Lock/LockSystem.cs:89]: Doesn't this mean the popup won't play for other players now?

File: Content.Shared/Storage/EntitySystems/SharedEntityStorageSystem.cs [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+    [Dependency] private readonly INetManager _net = default!;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly PlaceableSurfaceSystem _placeableSurface = default!;
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedContainerSystem _container = default!;
+    [Dependency] private readonly SharedInteractionSystem _interaction = default!;
+    [Dependency] private readonly SharedPhysicsSystem _physics = default!;
+    [Dependency] protected readonly SharedPopupSystem Popup = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
```
Comment (metalgearsloth) [inline Content.Shared/Storage/EntitySystems/SharedEntityStorageSystem.cs:38]: There's dupes with server system here.

File: Content.Server/Storage/Components/EntityStorageComponent.cs (hunk 1 above)
Comment (EmoGarbage404) [reply Content.Server/Storage/Components/EntityStorageComponent.cs:9]: nope, atmos mixtures are server-only.

File: Content.Shared/Storage/EntitySystems/SharedEntityStorageSystem.cs (hunk 3 above)
Comment (EmoGarbage404) [reply Content.Shared/Storage/EntitySystems/SharedEntityStorageSystem.cs:38]: there aren't, from what I can tell. Server only has atmos, construction, and Imapmanager.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**